### PR TITLE
Upgrade tonic+prost, unify dependencies, add lockfile

### PR DESCRIPTION
**Pull request by onalante-msft**
Also add serde derivations to google.protobuf.Struct. Tests for whether the derivations are complete are pending.

Adding the workspace lockfile for CI.

### CODE REVIEW

**Review by huguesBouvier — COMMENTED** [review]
Ah nice I see they updated! Thank you! You can now do this: let struct_obj: Struct = serde_json::from_str(struct_json.as_str()).unwrap();
Much better....

I also see that you fixed the tempdir bug (Getting out of scope too early).

Fix the broken test and I approve :).

**Comment by onalante-msft** [comment]
> Ah nice I see they updated! Thank you! You can now do this: let struct_obj: Struct = serde_json::from_str(struct_json.as_str()).unwrap(); Much better....

Yes, it took some fiddling with the attributes: https://github.com/huguesBouvier/E4K/blob/0857e2032f2c96e79282c40a260aa7e63f7d512b/identities/common/workload-api/build.rs#L34-L39. It is not ideal (I would rather not have to tweak codegen for builtins), but it allows us to use a single version of proc-macro2/syn/quote.

> Fix the broken test and I approve :).

The build failures are due to missing a protobuf compiler. The Debian and Ubuntu versions we are testing can install a recent-enough protobuf compiler (or build one from source), but CentOS 7 lacks both a recent enough protobuf compiler version and a recent enough CMake version to build a protobuf compiler. Should we also build CMake from source for CentOS so we can get a recent protobuf compiler built?

File: identities/iot-edge-spiffe-agent/agentd/Cargo.toml [hunk 1]
```diff
@@ -8,13 +8,13 @@ edition = "2021"
 [dependencies]
 async-stream = "0.3"
 futures-util = "0.3"
-kube = { version = "0.69.0", features = ["runtime", "derive"] }
+kube = { version = "0.70.0", features = ["runtime", "derive"] }
```
Comment (huguesBouvier) [inline identities/iot-edge-spiffe-agent/agentd/Cargo.toml:11]: NP: 0.70

**Review by huguesBouvier — APPROVED** [review]

**Review by huguesBouvier — COMMENTED** [review]
Thanks for the clean up too!

**Pull request merged**